frontend/apps/filemanager/filemanagerhistory.lua

```diff
@@ -154,17 +162,98 @@ function FileManagerHistory:onShowHist()
         covers_fullscreen = true, -- hint for UIManager:_repaint()
         is_borderless = true,
         is_popout = false,
+        title_bar_left_icon = "appbar.menu",
+        onLeftButtonTap = function() self:showHistDialog() end,
         onMenuHold = self.onMenuHold,
         onSetRotationMode = self.MenuSetRotationModeHandler,
         _manager = self,
     }
 
     self:updateItemTable()
     self.hist_menu.close_callback = function()
+        self.status = nil
+        self.filter = nil
         UIManager:close(self.hist_menu)
     end
     UIManager:show(self.hist_menu)
     return true
 end
 
+function FileManagerHistory:showHistDialog()
+    if not self.status then
+        local _status
```

Comment:
OK, got confused by the name `showHistDialog()` which I thought was opening the History itself.

May be give self.status another name, more different than your _status.
`self.has_cached_status    self.has_statuses    self.statuses_fetched`...